connectors/byoc.py

```diff
@@ -578,39 +511,42 @@ def next_sync(self):
         if self.sync_now:
             logger.debug("sync_now is true, syncing!")
             return 0
-        if not self.scheduling["enabled"]:
+        if not self.scheduling.get("enabled", False):
             logger.debug("scheduler is disabled")
             return SYNC_DISABLED
-        return next_run(self.scheduling["interval"])
+        return next_run(self.scheduling.get("interval"))
+
+    async def reset_sync_now_flag(self):
+        await self.index.update(doc_id=self.id, doc={"sync_now": False})
+
+    async def sync_starts(self):
+        doc = {
+            "last_sync_status": JobStatus.IN_PROGRESS.value,
+            "last_sync_error": None,
+            "status": Status.CONNECTED.value,
+        }
+        await self.index.update(doc_id=self.id, doc=doc)
 
     async def _sync_starts(self, sync_job_index):
```

Comment:
`_sync_starts` will eventually be removed once job execution is moved out of `Connector` class in a separate PR.

Reply:
Can you add a `TODO` ?

Reply:
Done in https://github.com/elastic/connectors-python/pull/474/commits/1224df3fe0836512deffca22102671570b8ca9a7